packages/@aws-cdk/aws-codecommit/test/codecommit.test.ts

```diff
@@ -66,6 +70,91 @@ describe('codecommit', () => {
 
     });
 
+    test('can be setup with file deployment', () => {
+      // GIVEN
+      const app = new App({
+        context: {
+          [cxapi.DISABLE_ASSET_STAGING_CONTEXT]: 'true',
+        },
+      });
+      const stack = new Stack(app, 'MyStack');
+
+      // WHEN
+
+      new Repository(stack, 'Repository', {
+        repositoryName: 'MyRepositoryName',
+        code: Code.fromZipFile(join(__dirname, 'asset-test.zip')),
+      });
+
+      // THEN
+      const entry = stack.node.metadataEntry.find(m => m.type === 'aws:cdk:asset');
+      expect(entry).toBeTruthy();
+
+      // verify that now the template contains parameters for this asset
+      const assembly = app.synth();
+      const template = JSON.parse(readFileSync(join(assembly.directory, 'MyStack.template.json'), { encoding: 'utf-8' }));
+
+      expect(template.Resources.Repository22E53BBD.Properties.Code.S3.Bucket).toBeDefined();
```

Comment:
For the tests, I meant something like this: https://github.com/aws/aws-cdk/blob/203c42b792e46536eaf21fa48277985c3409b65c/packages/%40aws-cdk/aws-s3-assets/test/asset.test.ts#L248-L248

And assert you get one Asset back.

Same comment in the below tests.